testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1282,7 +1301,7 @@ impl BitcoinRegtestController {
 
     /// wait until the ChainsCoordinator has processed sortitions up to the
     ///   canonical chain tip, or has processed up to height_to_wait
-    pub fn wait_for_sortitions(&self, height_to_wait: Option<u64>) -> BurnchainTip {
+    pub fn wait_for_sortitions(&self, height_to_wait: Option<u64>) -> Option<BurnchainTip> {
```

Comment:
Should this be a `Result<BurnchainTip, burnchain_error>` return type now?  Returning `None` tells the caller less information than `Err(burnchain_error::CoordinatorClosed)` would, and leaves space for more error types.  What do you think?

Reply:
Yeah I thought about this, and I'll unpack the rationale I had in mind.
I personally believe that if we're going for `stacks-node` rewrite, we should be embracing a Runloop design pattern, where the main thread is responsible, and only responsible for orchestrating actions across the different threads, versus having a myriad of add-hoc channels between threads.
I acknowledge the fact that the overall approach could look patchy at first sight, but my goal was to ensure that thread terminations are orchestrated by the main thread, vs the main thread terminating the coordinator thread, implicitly terminating other threads.

Reply:
That's a great idea!  My feedback here was written with this in mind.  If we're going to structure the threads controlled by the runloop such that they have designated "preemption points" whereby they poll on a shared flag to see if the stop signal has been given, then it still makes sense to have various methods report an error condition when the preemption point has been reached and the shared flag indicates that the method should return.  This is largely how all existing systems today handle this -- even POSIX uses the `EINTR` error code that gets set by blocking I/O syscalls so that main loops can at least know that the syscall was aborted for this particular reason.

Reply:
Addressed with latest